src/sunmatrix/sparse/sunmatrix_sparse.c

```diff
@@ -1163,22 +1206,26 @@ int Matvec_SparseCSC(SUNMatrix A, N_Vector x, N_Vector y)
  *
  * Returns 0 if successful, -1 if unsuccessful (failed memory access).
  */
-int Matvec_SparseCSR(SUNMatrix A, N_Vector x, N_Vector y)
+SUNErrCode Matvec_SparseCSR(SUNMatrix A, N_Vector x, N_Vector y)
 {
   sunindextype i, j;
   sunindextype *Ap, *Aj;
   sunrealtype *Ax, *xd, *yd;
+  SUNFunctionBegin(A->sunctx);
 
   /* access data from CSR structure (return if failure) */
   Ap = SM_INDEXPTRS_S(A);
+  SUNCheck(Ap, SUN_ERR_ARG_CORRUPT);
   Aj = SM_INDEXVALS_S(A);
+  SUNCheck(Aj, SUN_ERR_ARG_CORRUPT);
   Ax = SM_DATA_S(A);
-  if ((Ap == NULL) || (Aj == NULL) || (Ax == NULL)) { return SUNMAT_MEM_FAIL; }
+  SUNCheck(Ax, SUN_ERR_ARG_CORRUPT);
 
   /* access vector data (return if failure) */
   xd = N_VGetArrayPointer(x);
+  SUNCheckLastErr();
   yd = N_VGetArrayPointer(y);
-  if ((xd == NULL) || (yd == NULL) || (xd == yd)) { return SUNMAT_MEM_FAIL; }
+  SUNCheckLastErr();
 
```

Comment:
We previously checked that `xd` and `yd` were non-NULL, and that they didn't point to the same memory.  Those checks have now been removed.
```suggestion
   SUNAssert(xd, SUN_ERR_ARG_CORRUPT);
   SUNAssert(yd, SUN_ERR_ARG_CORRUPT);
   SUNAssert(xd != yd, SUN_ERR_ARG_CORRUPT);
```